src/types/composer.ts

```diff
@@ -582,6 +266,17 @@ export class TransactionComposer {
 
   private errorTransformers: ErrorTransformer[]
 
+  private composerConfig: TransactionComposerConfig
+
+  private transactionsWithSigners?: TransactionWithSigner[]
+
+  private signedTransactions?: SignedTransaction[]
+
+  // Note: This doesn't need to be a private field of this class
+  // It has been done this way so that another process can manipulate this values, i.e. `legacySendTransactionBridge`
+  // Once the legacy bridges are removed, this can be calculated on the fly
+  private methodCalls: Map<number, algosdk.ABIMethod> = new Map()
```

Comment:
Having to keep this in sync isn't great. I'm wondering if we could instead calculate on the fly (as you mentioned above), however add a private method which the legacy bridge can call?